internal/provider/resource_password.go

```diff
@@ -47,9 +47,256 @@ func (r resourcePassword) Delete(ctx context.Context, req tfsdk.DeleteResourceRe
 }
 
 func (r resourcePassword) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
-	importString(ctx, req, resp, true)
+	importPassword(ctx, req, resp)
 }
 
 func (r resourcePassword) ValidateConfig(ctx context.Context, req tfsdk.ValidateResourceConfigRequest, resp *tfsdk.ValidateResourceConfigResponse) {
 	validateLength(ctx, req, resp)
 }
+
+func (r resourcePassword) UpgradeState(context.Context) map[int64]tfsdk.ResourceStateUpgrader {
+	return map[int64]tfsdk.ResourceStateUpgrader{
+		0: {
+			StateUpgrader: migratePasswordStateV0toV1,
+		},
```

Comment:
Any particular reason to opt for handling `RawState` in the `StateUpgrader` versus specifying the version 0 schema here in the `PriorSchema` field? If you use `PriorSchema`, you can call `req.PriorState.Get()` to populate the already defined data type.

Reply:
Yep, I went the route of using `RawState` as `req.State` is nil in the `tfsdk.UpgradeResourceStateRequest` and I can't see a `PriorState` field in `req`.

Reply:
Sorry I meant, [`UpgradeResourceStateRequest.State`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#UpgradeResourceStateRequest). It get's filled in when [`ResourceStateUpgrader.PriorSchema`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#ResourceStateUpgrader) is defined. It should make the logic a lot easier.

Reply:
Thanks for the clarification. I've refactored to use `StateUpgrader` with `PriorSchema`.